python-threatexchange/threatexchange/extensions/vpdq/tests/test_vpdq_faiss.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+from cmath import sin
+import pytest
+from pathlib import Path
+import pickle
+
+try:
+    import vpdq as _
+
+    _DISABLED = False
+except (ImportError, ModuleNotFoundError) as e:
+    _DISABLED = True
+else:
+    import typing as t
+    from threatexchange.extensions.vpdq.vpdq_index import VPDQIndex
+    from threatexchange.extensions.vpdq.vpdq_util import (
+        prepare_vpdq_feature,
+        VPDQ_QUALITY_THRESHOLD,
+    )
+    from threatexchange.extensions.vpdq.video_vpdq import VideoVPDQSignal
+
+    from threatexchange.signal_type.index import (
+        VPDQIndexMatch,
+    )
+
+ROOTDIR = Path(__file__).parents[5]
+example_meta_data = {"name": "example_video"}
+
+
+@pytest.mark.skipif(_DISABLED, reason="vpdq not installed")
+def test_simple():
+    hash = VideoVPDQSignal.get_examples()[0]
+    features = prepare_vpdq_feature(hash, VPDQ_QUALITY_THRESHOLD)
+    index = VPDQIndex.build([[hash, example_meta_data]])
+    assert index._entry_idx_to_features_and_entires[0][0] == features
+    assert len(index._index_idx_to_vpdqHex_and_entry) == len(features)
+    res = index.query(hash)
+    # A complete match to itself
+    assert compare_match_result(res[0], VPDQIndexMatch(-1, 100, 100, example_meta_data))
+
+
+@pytest.mark.skipif(_DISABLED, reason="vpdq not installed")
+def test_serialize():
+    hash = VideoVPDQSignal.get_examples()[0]
+    index = VPDQIndex.build([[hash, example_meta_data]])
+    pickled_data = pickle.dumps(index)
+    reconstructed_index = pickle.loads(pickled_data)
+    res = reconstructed_index.query(hash)
+    assert compare_match_result(res[0], VPDQIndexMatch(-1, 100, 100, example_meta_data))
+
+
+@pytest.mark.skipif(_DISABLED, reason="vpdq not installed")
+def test_duplicate_hashes():
+    hash = VideoVPDQSignal.get_examples()[0]
+    index = VPDQIndex.build([[hash, "video1"]])
+    index.add(hash, "video2")
+    res = index.query(hash)
+    # A complete match to itself
+    assert compare_match_result(res[0], VPDQIndexMatch(-1, 100, 100, "video1"))
+    assert compare_match_result(res[1], VPDQIndexMatch(-1, 100, 100, "video2"))
```

Comment:
All your tests are all asserting that they are perfect matches, which means that many failure modes might be missed. I am not yet convinced that your implementation is correct as a result. Would like to see some variation here in a followup PR.

i.e.
```
pdq_dist(A, A) = 0
pdq_dist(A, B) = 31
pdq_dist(A, C-F) > 31

v1 = [A]
v2 = [A, B]
v3 = [B]
V4 = [B, C]
V5 = [B, C, D]

for x in range(1, 6):
  for y in range(x, 6):
    vpdq_dist(v{x}, v{y})
    vpdq_dist(V{y}, V{x})

```